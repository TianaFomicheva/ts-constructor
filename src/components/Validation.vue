<template>
    <Switch />
</template>

<script lang="ts">
import { defineComponent } from 'vue';
import Switch from '@/elements/Switch.vue';

export default defineComponent({
  name: 'validation',
  components: {
    Switch,
  },
});
</script>

<style lang="scss">
</style>
