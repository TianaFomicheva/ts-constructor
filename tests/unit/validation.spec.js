import { shallowMount } from '@vue/test-utils';
import Validation from '@/components/Validation.vue';

describe('Validation.vue', () => {
  it('renders title', () => {
    const msg = 'Валидация';
    const wrapper = shallowMount(Validation, {
      props: { msg },
    });
    expect(wrapper.text()).toMatch(msg);
  });
  it('renders snapshot', () => {
    const wrapper = shallowMount(Validation);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
