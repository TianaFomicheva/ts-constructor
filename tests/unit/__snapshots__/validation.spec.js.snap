// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validation.vue renders snapshot 1`] = `
<h3 class="title">Валидация</h3>
<div class="form-field">
  <switch-stub propval="false"></switch-stub>
</div>
<div class="form-field">
  <min-max-stub prop-val="[object Object]"></min-max-stub>
</div>
<!--v-if-->
`;
